tests/returntypes/test_autograd_qnode_shot_vector.py

```diff
@@ -0,0 +1,687 @@
+# Copyright 2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Integration tests for using the jax interface with shot vectors and with a QNode"""
+
+import pytest
+
+import pennylane as qml
+from pennylane import numpy as np
+from pennylane import qnode
+
+pytestmark = pytest.mark.autograd
+
+shots = [((5, 2), 1, 10)]
+
+qubit_device_and_diff_method = [
+    ["default.qubit", "finite-diff", {"h": 10e-2}],
+    ["default.qubit", "parameter-shift", {}],
+]
+
+
+@pytest.mark.parametrize("shots", shots)
+@pytest.mark.parametrize("dev_name,diff_method,gradient_kwargs", qubit_device_and_diff_method)
+class TestReturnWithShotVectors:
+    """Class to test the shape of the Grad/Jacobian/Hessian with different return types and shot vectors."""
+
+    def test_jac_single_measurement_param(self, dev_name, diff_method, gradient_kwargs, shots):
+        """For one measurement and one param, the gradient is a float."""
+        dev = qml.device(dev_name, wires=1, shots=shots)
+
+        @qnode(dev, diff_method=diff_method, **gradient_kwargs)
+        def circuit(a):
+            qml.RY(a, wires=0)
+            qml.RX(0.2, wires=0)
+            return qml.expval(qml.PauliZ(0))
+
+        a = np.array(0.1)
+
+        def cost(a):
+            return qml.math.stack(circuit(a))
+
+        jac = qml.jacobian(cost)(a)
```

Comment:
1. Single return with jacobian.

Reply:
Do these comments refer to action items, or are they just for bookkeeping?

Reply:
I think that the four different cases that users should know about and that should be documented, because they have different ways of stacking for it to work.

Reply:
Added a new shot vectors section to the documentation in `qml.enable_return`.